Remotion/Web/Development.WebTesting/WebDriver/Factories/Chrome/ChromeBrowserFactory.cs

```diff
@@ -46,6 +46,9 @@ public IBrowserSession CreateBrowser (DriverConfiguration configuration)
     {
       ArgumentUtility.CheckNotNull ("configuration", configuration);
 
+      var registryCleanUpFactory = new ChromeSecurityWarningsRegistryCleanUpStrategyFactory (_chromeConfiguration.DisableSecurityWarningsBehavior);

```

Comment:
Should we move this to the ctor? 

In theory, we could also do dependency injection but I'm not 100% this is the best design decision: Does it fit with the exisitng parameters?  Who's responsible for composing the object graph? Is the usage of the cleanup factory a mandatory part of the chrome browser factory?

Reply:
> Should we move this to the ctor?

Good point, will do

> In theory, we could also do dependency injection but I'm not 100% this is the best design decision: Does it fit with the exisitng parameters? Who's responsible for composing the object graph? Is the usage of the cleanup factory a mandatory part of the chrome browser factory?

* `ChromeBrowserFactory` has `ChromeConfiguration` as ctor parameter.
* `ChromeBrowserFactory` is initialized in `ChromeConfiguration`
* I do think that the `ChromeSecurityWarningsRegistryCleanUpStrategyFactory` is a mandatory part of the `ChromeBrowserFactory`. The `ChromeBrowserFactory` is responsible for creating the WebDriver instance, configured using the injected `ChromeConfiguration`. Setting the needed registry values is part of the browser/webdriver creation process. Besides, there currently is no need for an interface, no alternative `ChromeSecurityWarningsRegistryCleanUpStrategyFactory` would make sense here and unit testing the `ChromeBrowserFactory` would be a pain anyway.

Reply:
+1 to moving the factory to the ctor, -1 to injecting it. I don't currently see any advantage for DI at this point, see @iam-mholle 's comments

Reply:
sold